outlines/samplers.py

```diff
@@ -66,7 +66,9 @@ def __call__(
         logprobs = torch.nn.functional.log_softmax(next_token_logits, dim=-1)
         next_token_ids = torch.argmax(logprobs, dim=-1, keepdim=True)
 
-        ancestors = torch.arange(next_token_logits.shape[0])
+        ancestors = torch.arange(
+            next_token_logits.shape[0], device=next_token_logits.device
```

Comment:
nit: let's be consistent across samplers and use `next_token_logits` for each sampler :)